pkg/reconciler/taskrun/taskrun.go

```diff
@@ -402,18 +405,23 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1beta1.TaskRun, rtr *re
 		// List pods that have a label with this TaskRun name.  Do not include other labels from the
 		// TaskRun in this selector.  The user could change them during the lifetime of the TaskRun so the
 		// current labels may not be set on a previously created Pod.
-		labelSelector := fmt.Sprintf("%s=%s", pipeline.TaskRunLabelKey, tr.Name)
-		pos, err := c.KubeClientSet.CoreV1().Pods(tr.Namespace).List(ctx, metav1.ListOptions{
-			LabelSelector: labelSelector,
-		})
+		labelSelector := labels.Everything()
+		myTaskLabelSelector, err := labels.NewRequirement(pipeline.TaskRunLabelKey, selection.Equals, []string{tr.Name})
+		if err != nil {
+			logger.Errorf("Failed to construct a label selector for %s = %s", pipeline.TaskRunLabelKey, tr.Name)
+			return err
+		}
+		labelSelector = labelSelector.Add(*myTaskLabelSelector)
```

Comment:
I think you can do something tighter here with `labels.SelectorFromSet` where `labels.Set` is just a `map[string]string` where you can say `labels.Set{pipeline.TaskRunLabelKey: tr.Name}` and avoid the error path above entirely.